### PR TITLE
Push notification keepalive connection handling

### PR DESCRIPTION
**Pull request by javrudsky**
# Android SDK

## What did you accomplish?
- Added keepalive to SSE client
- Added push manager tests

### CODE REVIEW

File: src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java [hunk 1]
```diff
@@ -52,6 +63,13 @@ private void connectToSse(String token, List<String> channels) {
         mSseClient.connect(token, channels);
     }
 
+    private void scheduleSseDownNotification() {
```
Comment (mredolatti) [inline src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java:66]: Can we rename this to `resetSSEKeepAliveTimer` or similar?

File: src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java [hunk 2]
```diff
@@ -60,60 +78,87 @@ private void notifyPushDisabled() {
...
+                Logger.e("Sse authentication error. Token or Channels not available.");
             }
-            return null;
+        } else {
+            Logger.e("Couldn't connect to SSE server. Streaming is disabled.");
+        }
+        return null;
+    }
+
+    private class SseDownNotificator implements SplitTask {
```
Comment (mredolatti) [inline src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java:155]: `SSEKeepAliveTimer`

File: src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java (hunk 1 above)
Comment (javrudsky) [reply src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java:66]: Sure. Done!

File: src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java (hunk 2 above)
Comment (javrudsky) [reply src/main/java/io/split/android/client/service/sseclient/PushNotificationManager.java:155]: Sure. Done!

**Pull request merged**